### PR TITLE
Zero-argument variants for table functions numbers, numbers_mt, zeros, and zeros_mt 

### LINKED ISSUE #61934: Zero-argument variants for table functions `numbers`, `numbers_mt`, `zeros`, and `zeros_mt`
**Use case**

Convenience.

**Describe the solution you'd like**

`SELECT * FROM numbers()` will work in the same way as `SELECT * FROM system.numbers` - without a limit.

### PR DESCRIPTION
**Pull request by YenchangChan**
Closes #61934 
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
For convenience purpose, `SELECT * FROM numbers() `will work in the same way as `SELECT * FROM system.numbers` - without a limit.

### Documentation entry for user-facing changes

For convenience purpose, `SELECT * FROM numbers() `will work in the same way as `SELECT * FROM system.numbers` - without a limit, which applies for table functions numbers, numbers_mt, zeros, and zeros_mt.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by jsc0218** [comment]
thanks for contributing, should add a test for new feature.

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 25d173917469845345247fcca9a7b19575cd02dc with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61969/25d173917469845345247fcca9a7b19575cd02dc/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

File: src/TableFunctions/TableFunctionZeros.cpp [hunk 1]
```diff
@@ -55,15 +55,24 @@ StoragePtr TableFunctionZeros<multithreaded>::executeImpl(const ASTPtr & ast_fun
     {
         auto arguments = function->arguments->children;
 
-        if (arguments.size() != 1)
+        if (arguments.size() >= 2)
             throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Table function '{}' requires 'length'.", getName());
```
Comment (jsc0218) [inline src/TableFunctions/TableFunctionZeros.cpp:59]: same as before

File: src/TableFunctions/TableFunctionNumbers.cpp [hunk 2]
```diff
@@ -70,18 +70,27 @@ StoragePtr TableFunctionNumbers<multithreaded>::executeImpl(
...
-
-        UInt64 offset = arguments.size() >= 2 ? evaluateArgument(context, arguments[0]) : 0;
-        UInt64 length = arguments.size() >= 2 ? evaluateArgument(context, arguments[1]) : evaluateArgument(context, arguments[0]);
-        UInt64 step = arguments.size() == 3 ? evaluateArgument(context, arguments[2]) : 1;
-
-        auto res = std::make_shared<StorageSystemNumbers>(
-            StorageID(getDatabaseName(), table_name), multithreaded, std::string{"number"}, length, offset, step);
-        res->startup();
-        return res;
+        if (arguments.size() > 0)
```
Comment (jsc0218) [inline src/TableFunctions/TableFunctionNumbers.cpp:76]: clang tidy complains here, it suggests to use `!arguments.empty()`, otherwise not mergeable.

**Comment by jsc0218** [comment]
@YenchangChan kindly ping you.

**Comment by YenchangChan** [comment]
> @YenchangChan kindly ping you.

Sorry, I've been a bit busy lately and I'll finish it over the weekend.

**Review by jsc0218 — APPROVED** [review]

**Comment by ramazanpolat** [comment]
Sorry, can someone explain the use case of this?

**Pull request merged**